plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/customization/CodeWhispererModelConfigurator.kt

```diff
@@ -165,20 +167,23 @@ class DefaultCodeWhispererModelConfigurator : CodeWhispererModelConfigurator, Pe
     override fun activeCustomization(project: Project): CodeWhispererCustomization? {
         val selectedCustomization = calculateIfIamIdentityCenterConnection(project) { connectionIdToActiveCustomizationArn[it.id] }
 
-        if (selectedCustomization != null) {
-            return selectedCustomization
-        } else {
-            val customizationOverride = CodeWhispererFeatureConfigService.getInstance().getCustomizationFeature()
-            if (customizationOverride == null || customizationOverride.value.stringValue().isEmpty()) return null
-            return CodeWhispererCustomization(
-                arn = customizationOverride.value.stringValue(),
-                name = customizationOverride.variation,
-            )
-        }
+        return selectedCustomization
     }
 
     override fun switchCustomization(project: Project, newCustomization: CodeWhispererCustomization?) {
+        switchCustomization(project, newCustomization, false)
+    }
+
+    /**
+     * Override happens when ALL following conditions are met
+     *  1. service returns non-empty override customization arn, refer to [CodeWhispererFeatureConfigService]
+     *  2. the override customization arn is different from the previous override customization if any. The purpose is to only do override once on users' behalf.
+     */
+    override fun switchCustomization(project: Project, newCustomization: CodeWhispererCustomization?, isOverride: Boolean) {
```

Comment:
might be trying to shove too much logic into this method

Reply:
yea, originally my plan was to simply call `switchCustomization` within `featureConfigService`, but there is dependency issue as modelConfigurator is still in codewhisperer package.

Reply:
think i sohuld move modelConfig to shared/ then we can remove this